freqtrade/commands/build_config_commands.py

```diff
@@ -0,0 +1,193 @@
+from freqtrade.exceptions import OperationalException
+logger = logging.getLogger(__name__)
+
+
+def validate_is_int(val):
+    try:
+        _ = int(val)
+        return True
+    except Exception:
+        return False
+
+
+def validate_is_float(val):
+    try:
+        _ = float(val)
+        return True
+    except Exception:
+        return False
+
+
+def ask_user_overwrite(config_path: Path) -> bool:
+    questions = [
+        {
+            "type": "confirm",
+            "name": "overwrite",
+            "message": f"File {config_path} already exists. Overwrite?",
+            "default": False,
+        },
+    ]
+    answers = prompt(questions)
+    return answers['overwrite']
+
+
+def ask_user_config() -> Dict[str, Any]:
+    """
+    Ask user a few questions to build the configuration.
+    Interactive questions built using https://github.com/tmbo/questionary
+    :returns: Dict with keys to put into template
+    """
+    questions = [
+        {
+            "type": "confirm",
+            "name": "dry_run",
+            "message": "Do you want to enable Dry-run (simulated trades)?",
+            "default": True,
+        },
+        {
+            "type": "text",
+            "name": "stake_currency",
+            "message": "Please insert your stake currency:",
+            "default": 'BTC',
+        },
+        {
+            "type": "text",
+            "name": "stake_amount",
+            "message": "Please insert your stake amount:",
+            "default": "0.01",
+            "validate": lambda val: val == UNLIMITED_STAKE_AMOUNT or validate_is_float(val),
+        },
+        {
+            "type": "text",
+            "name": "max_open_trades",
+            "message": f"Please insert max_open_trades (Integer or '{UNLIMITED_STAKE_AMOUNT}'):",
+            "default": "3",
+            "validate": lambda val: val == UNLIMITED_STAKE_AMOUNT or validate_is_int(val)
+        },
+        {
+            "type": "text",
+            "name": "ticker_interval",
+            "message": "Please insert your ticker interval:",
+            "default": "5m",
+        },
+        {
+            "type": "text",
+            "name": "fiat_display_currency",
+            "message": "Please insert your display Currency (for reporting):",
+            "default": 'USD',
+        },
+        {
+            "type": "select",
+            "name": "exchange_name",
+            "message": "Select exchange",
+            "choices": [
+                "binance",
+                "binanceje",
+                "binanceus",
+                "bittrex",
+                "kraken",
+                Separator(),
+                "other",
+            ],
+        },
+        {
+            "type": "autocomplete",
+            "name": "exchange_name",
+            "message": "Type your exchange name (Must be supported by ccxt)",
+            "choices": available_exchanges(),
+            "when": lambda x: x["exchange_name"] == 'other'
```

Comment:
I would move selection of the exchange to be the first question: 1) that's the most important question in the set imho, 2) selection of stake_currency depend on it; 3) possibly in the future we'll ask different set of questions depending on exchange used.

Reply:
To explain my thoughts on this:
* ask questions requiring keys / tokens only if required

Therefore, to me, dry-run needs to be first (or at least before exchange), as that determines if we need exchange key / token at all.
To however skip fro dry-run to exchange and then "back" to the simple questions (stake / stake amount / max_open_trades) seemed strange when trying it out.

Also (and i consider this a bug/ missing feature in questionaire) - boolean questions behave odd (to me) in providing an "auto-enter" when "Y" / "n" is pressed - so by asking a "simple" question right after that where the default can easily be changed afterwards - we can avoid to have to go through the flow multiple times.

Reply:
Selecting exchange is also a "simple" question -- and the main one in my opinion. And answering it first is straightforward for a user -- first select exchange, then all supplicant questions (simple and more complex) follow...

> To however skip fro dry-run to exchange and then "back" to the simple questions (stake / stake amount / max_open_trades) seemed strange when trying it out.

right, but if you move selection of exchange at the top, you won't need to skip...
Select exchange, then the mode (dry run/live) *on that exchange*... 

A user may want to have live trade mode on an exchange she has an account and dry-run mode on other exchanges, so the mode *depends* on the exchange selected...

Reply:
That doesn't make sense ...

exchange and api-key / api-token are closely related. 
I will not split these up - that would be pointless and hopelessly confusing to users - and with that dry-run will stay before exchange.

If we're honest - we can also ask this differently:

* do you want to simulate trades?
* how many trades?
* in which stake currency?
* and on what exchange?

As a user, you'll need to know all answers anyway - or at least think about them while answering. 
The sequence of the questions does not create a different answer.

Reply:
In my opinion the answer on "do you want to simulate trades?" also depends on the exchenge, selected, but... let it be as it is for now...

LGTM 👍